mne/beamformer/_compute_beamformer.py

```diff
@@ -166,6 +107,11 @@ def _prepare_beamformer_input(info, forward, label, picks, pick_ori,
         raise ValueError('Normal orientation can only be picked when a '
                          'forward operator with a surface-based source space '
                          'is used.')
+    # Check whether data and forward model have same compensation applied
+    if not get_current_comp(info) == get_current_comp(forward['info']):
+        raise ValueError('Data and forward model do not have same '
+                         'compensation applied.')
+
```

Comment:
@agramfort : this is new, checking whether compensation is identical between data and forward model